libs/server/controllers/friend/src/lib/friend.controller.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  HasRole,
+  RoleGuard,
+  GetCurrentUser,
+  RemoveSensitiveFieldsInterceptor,
+} from '@bregenz-bewegt/server/common';
+import {
+  SearchUserQueryDto,
+  FriendSearchResult,
+  CreateFriendRequestDto,
+  AllFriendRequests,
+  FriendAdresseeResult,
+  FriendRequesteeResult,
+  RevokeFriendRequestDto,
+  RejectFriendRequestDto,
+  AcceptFriendRequestDto,
+  RemoveFriendDto,
+} from '@bregenz-bewegt/shared/types';
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Post,
+  Query,
+  UseGuards,
+  UseInterceptors,
+  ValidationPipe,
+} from '@nestjs/common';
+import { Role, User, FriendRequest } from '@prisma/client';
+import { FriendService } from './friend.service';
+
+@Controller('friends')
+export class FriendController {
+  constructor(private friendService: FriendService) {}
+
+  @HasRole(Role.USER)
+  @UseGuards(RoleGuard)
+  @Get()
+  getFriends(@GetCurrentUser('sub') userId: User['id']): Promise<User[]> {
+    return this.friendService.getFriends(userId);
+  }
+
+  @HasRole(Role.USER)
+  @UseGuards(RoleGuard)
+  @UseInterceptors(RemoveSensitiveFieldsInterceptor)
+  @Get('search')
+  searchUser(
+    @GetCurrentUser('sub') userId: User['id'],
+    @Query(
+      new ValidationPipe({
+        transform: true,
+        transformOptions: { enableImplicitConversion: true },
+        forbidNonWhitelisted: true,
+      })
+    )
+    dto: SearchUserQueryDto
+  ): Promise<FriendSearchResult[]> {
+    return this.friendService.searchUserFriendByUsername(dto.username, userId);
+  }
+
+  @HasRole(Role.USER)
+  @UseGuards(RoleGuard)
+  @Delete('remove')
+  removeFriend(
+    @GetCurrentUser('sub') userId: User['id'],
+    @Body() dto: RemoveFriendDto
+  ): Promise<User> {
+    return this.friendService.removeFriend(userId, dto);
+  }
+
+  @HasRole(Role.USER)
+  @UseGuards(RoleGuard)
+  @Post('request')
+  createFriendRequest(
+    @GetCurrentUser('sub') requesteeId: User['id'],
+    @Body() dto: CreateFriendRequestDto
+  ): Promise<FriendRequest> {
+    return this.friendService.createFriendRequest({
+      requesteeId,
+      addresseeId: dto.addresseeId,
+    });
+  }
+
+  @HasRole(Role.USER)
+  @UseGuards(RoleGuard)
+  @UseInterceptors(RemoveSensitiveFieldsInterceptor)
+  @Get('requests')
+  async getFriendRequests(
+    @GetCurrentUser('sub') userId: User['id']
+  ): Promise<AllFriendRequests> {
+    const requested = await this.friendService.getRequestedFriendRequests(
+      userId
+    );
+    const received = await this.friendService.getReceivedFriendRequests(userId);
+
+    return { requested, received };
+  }
+
+  @HasRole(Role.USER)
+  @UseGuards(RoleGuard)
+  @UseInterceptors(RemoveSensitiveFieldsInterceptor)
+  @Get('requests/requested')
+  getRequestedFriendRequests(
+    @GetCurrentUser('sub') userId: User['id']
+  ): Promise<(FriendRequest & { addressee: FriendAdresseeResult })[]> {
```

Comment:
bei FirendRequest sind no viele Felder dabei wo ma garnet brucht - i würd da client type FriendRequest vom Frontend in Shared tua und do verwenda anstatt der vo Prisma

Reply:
Wia musch des? Die frontend entity hot genau 2 felder:

```
export type FriendRequest = {
  id: string;
  createdAt: Date;
};
```

Reply:
ah des wollt i no in da Schual bespreacha, hobs vergeassa sry
I hob damit gmoant das die FriendRequest im Frontend, wie du gset hosch, nur 2 felder hot. Doch mir senden vom Backend us des ganze Friendrequest mit viel mehr Feldern mit was es eigentlich net brücht. Deswegen künnt ma an derara Stelle o der Typ FriendRequest vom Frontend nutza, glob i